src/neo/SmartContract/Native/GasToken.cs

```diff
@@ -40,7 +40,7 @@ internal override async ContractTask OnPersist(ApplicationEngine engine)
         [ContractMethod(CpuFee = 1 << 15, RequiredCallFlags = CallFlags.States | CallFlags.AllowNotify)]
         private async ContractTask Refuel(ApplicationEngine engine, UInt160 account, long amount)
         {
-            if (amount < 0) throw new ArgumentOutOfRangeException(nameof(amount));
+            if (amount <= 0) throw new ArgumentOutOfRangeException(nameof(amount));
             if (!engine.CheckWitnessInternal(account)) throw new InvalidOperationException();
             await Burn(engine, account, amount);
             engine.Refuel(amount);
```

Comment:
```suggestion
            engine.Refuel(amount);
            await Burn(engine, account, amount);
```

Reply:
What's the difference?

Reply:
We don't need do burn when refuel check failed.